CDCgov/prime-reportstream
# Create zipCodeToState Mapper for RADx

This PR adds a new mapper, ZipCodeToState, that utilizes the zip-code-data lookup table to add the patient state based on the zip code. Currently, the application will overwrite any incoming value based on the lookup results. To avoid overwriting a non-blank state value, this adds an additional field to the hl7-ingest-covid-19 schema, patient_state_zip_lookup. Additionally, this PR fixes a bug with the lookup mapper. The lookup mapper assumed that fields used in the mapper all used the same lookup table. That is an unsafe assumption. The PR fixes it so that the lookup columns are set in the parameters of the mapper removing the dependency.

Test Steps:
1. build and run all three test files through the application
     - `./prime data --input /PATH/TO/TEST/FILE.txt --input-schema HL7/hl7-ingest-covid-19-training  --output-dir /PATH/TO/OUTPUT/DIR/ --route`
2. Check output HL7
     - intrivo-test-missing-zip.txt: patient state in the output should be the same as input
     - intrivo-test-with-state.txt: patient state in the output should be same as input
     - intrivo-test-missing-state.txt: patient state in output should be different from input

Test Files
- [intrivo-test-missing-zip.txt](https://github.com/CDCgov/prime-reportstream/files/9237768/intrivo-test-missing-zip.txt)
- [intrivo-test-with-state.txt](https://github.com/CDCgov/prime-reportstream/files/9237769/intrivo-test-with-state.txt)
- [intrivo-test-missing-state.txt](https://github.com/CDCgov/prime-reportstream/files/9237770/intrivo-test-missing-state.txt)


## Changes
- added zipCodeToState mapper
- add column parameters for lookup mapper
- add unit test for zipCodeToState

## Checklist

### Testing

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #5700
